### PR TITLE
zcason/feature flag filter fix

### PR DESCRIPTION
**Pull request by zcason**
Signed-off-by: zcason <a-zcason@expediagroup.com>

## Hey, I just made a Pull Request!

I rewrote the feature flag filter functionality, it was broken previously.

feature-flag-fix

Screen Shot 2022-12-27 at 6 06 00 PM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Thanks! I think the filter was intended to work a bit differently before, but I think this makes sense tbh, is more in line with what one would expect, and is simpler.

Could you add a changeset as well, it's needed for this to be released? See the comment above for details.

Would be lovely to add some tests as well if you're in the mood :grin:

**Comment by zcason** [comment]
> Thanks! I think the filter was intended to work a bit differently before, but I think this makes sense tbh, is more in line with what one would expect, and is simpler.
> 
> Could you add a changeset as well, it's needed for this to be released? See the comment above for details.
> 
> Would be lovely to add some tests as well if you're in the mood 😁

Sure thing! I will ping you when once it's finished. Thank you for the feedback.

File: .changeset/loud-beds-occur.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-user-settings': major
```
Comment (Rugvip) [inline .changeset/loud-beds-occur.md:2]: Let's make this a `patch` bump. There's no API change and I'd consider the behavior change a fix.

**Review by Rugvip — CHANGES_REQUESTED** [review]

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 

Let's :shipit: 🎉

**Pull request merged**